apps/studio/components/interfaces/Settings/Infrastructure/InfrastructureConfiguration/EnablePhysicalBackupsModal.tsx

```diff
@@ -0,0 +1,143 @@
+import { ExternalLink, Loader2 } from 'lucide-react'
+import { useState } from 'react'
+import toast from 'react-hot-toast'
+
+import { useParams } from 'common'
+import { Markdown } from 'components/interfaces/Markdown'
+import { useEnablePhysicalBackupsMutation } from 'data/database/enable-physical-backups-mutation'
+import { useProjectDetailQuery } from 'data/projects/project-detail-query'
+import { Region, useReadReplicaSetUpMutation } from 'data/read-replicas/replica-setup-mutation'
+import { useReadReplicasQuery } from 'data/read-replicas/replicas-query'
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogSection,
+  DialogSectionSeparator,
+  DialogTitle,
+  DialogTrigger,
+} from 'ui'
+import { AVAILABLE_REPLICA_REGIONS, AWS_REGIONS_VALUES } from './InstanceConfiguration.constants'
+
+interface EnablePhysicalBackupsModalProps {
+  selectedRegion: string
+}
+
+export const EnablePhysicalBackupsModal = ({ selectedRegion }: EnablePhysicalBackupsModalProps) => {
+  const { ref } = useParams()
+  const [open, setOpen] = useState(false)
+  const [enabling, setEnabling] = useState(false)
+  const [refetchInterval, setRefetchInterval] = useState<number | false>(false)
+
+  const { data: databases } = useReadReplicasQuery({ projectRef: ref })
+
+  useProjectDetailQuery(
+    { ref },
+    {
+      refetchInterval,
+      refetchOnWindowFocus: false,
+      onSuccess: (data) => {
+        if (data.is_physical_backups_enabled) {
+          setRefetchInterval(false)
+          const regionKey = AWS_REGIONS_VALUES[selectedRegion]
+
+          if (ref === undefined) return toast.error('Project ref is required')
```

Comment:
feedback on our DX, I see this everywhere.
maybe just bake it into the react queries or something that doesn't mean we have to keep doing a undefined check anytime we want to make a request?

Reply:
yeah agree on this - its a bit odd this DX, will chat with alaister about this